third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1053,15 +1053,20 @@ fn check_depth_of_type_impl(
         | Type::Signer => check_depth!(0),
         // Even though this is recursive this is OK since the depth of this recursion is
         // bounded by the depth of the type arguments, which we have already checked.
-        Type::Reference(ty) | Type::MutableReference(ty) | Type::Vector(ty) => {
+        Type::Reference(ty) | Type::MutableReference(ty) => {
             check_depth_of_type_impl(resolver, ty, max_depth, check_depth!(1))?
         },
-        Type::Struct { name, .. } => {
-            let formula = resolver.loader().calculate_depth_of_struct(name)?;
+        Type::Vector(ty) => {
+            check_depth_of_type_impl(resolver, ty, max_depth, check_depth!(1))?
```

Comment:
Seems like the same line as above?

Reply:
The problem is that rust requires the same type for all matches. The reference has `&Box<Type>` whereas vector has `Arc<Type>`, so it cannot share the same matching clause.